launchers/generic/Dockerfile

```diff
@@ -0,0 +1,27 @@
+#
+#  Copyright (c) 2022 Microsoft Corporation
+#
+#  This program and the accompanying materials are made available under the
+#  terms of the Apache License, Version 2.0 which is available at
+#  https://www.apache.org/licenses/LICENSE-2.0
+#
+#  SPDX-License-Identifier: Apache-2.0
+#
+#  Contributors:
+#       Microsoft Corporation - initial API and implementation
+#
+#
+
+FROM openjdk:11-jre-slim
+ARG JAR
+
+RUN apt-get update && apt-get install -y \
+  bash \
+  && rm -rf /var/lib/apt/lists/*
+
+COPY $JAR app.jar
+
+# Use "bash" (instead of default "sh") to support environment variables containing a dot. 
+# Use "exec" for Kubernetes graceful termination (SIGINT) to reach JVM.
+ENTRYPOINT [ "bash", "-c", \
+    "exec java $JVM_ARGS -jar app.jar"]
```

Comment:
Do we really have environment variables with a dot? Thought we have some kind mapping mechanism which transforms between screaming snake case and the dot case. E.g `STUFF_TEST` to `stuff.test`

Reply:
This mapping exists only for settings fetched with `propOrEnv` e.g. the location of the config file `edc.fs.config`, not for other settings:

```
$ env web_http_port=9192 WEB_HTTP_PORT=9192 java -jar samples/04.0-file-transfer/consumer/build/libs/consumer.jar
INFO 2022-02-17T08:53:52.549321 HTTP context 'default' listening on port 8181


$ env web.http.port=9192 java -jar samples/04.0-file-transfer/consumer/build/libs/consumer.jar
INFO 2022-02-17T08:54:00.39245 HTTP context 'default' listening on port 9192
```

Reply:
Interesting. Do you or anyone else know if there is already a plan to align this to have some consistent ways to set the variables?

Reply:
In issue #608 there is a discussion to make environment variables always snake case.

Reply:
I think the following should happen:
- internally in the runtime we only use the dot notation
- the `Config` mechanism checks for properties in the dot notation and env-vars in the uppercase-underscore notation
- `propOrEnv` should not be used anywhere but tests

Reply:
to be fair, that conversion was not in place before, it has only been implemented in the `propOrEnv`. However an issue about that was already opened: #608

Reply:
Thank you guys! Didn't want to start a discussion here, so lets move over with this topic to the other issue 😉

Reply:
doesn't matter now, i didn't want to assign blame :) Your comment on #608 is correct and should be done so.

Reply:
How about fixing the setting properties by normalizing them as suggested by @ndr-brt and @paullatzelsperger and then realigning this PR with that?

Reply:
I'm OK to merge this as-is and fix the other issue ASAP. We can then convert shells in a subsequent PR if wanted.